bika/lims/browser/js/coffee/bika.lims.worksheet.coffee

```diff
@@ -0,0 +1,1070 @@
+    @load_analysis_method_constraint analysis_uid, method_uid
+
+
+  on_analysis_instrument_focus: (event) =>
+    ###
+     * Eventhandler when the instrument of an analysis is focused
+     *
+     * Only needed to remember the last value
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_analysis_instrument_focus °°°"
+
+    $el = $(event.currentTarget)
+    @previous_instrument = $el.val()
+    console.info @previous_instrument
+
+
+  on_analysis_instrument_change: (event) =>
+    ###
+     * Eventhandler when the instrument of an analysis changed
+     *
+     * If a new instrument is chosen for the analysis, disable this Instrument
+     * for the other analyses. Also, remove the restriction of previous
+     * Instrument of this analysis to be chosen in the other analyses.
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_analysis_instrument_change °°°"
+    $el = $(event.currentTarget)
+
+    analysis_uid = $el.attr "uid"
+    instrument_uid = $el.val()
+
+    # Disable New Instrument for rest of the analyses
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'][value!='#{instrument_uid}'] option[value='#{instrument_uid}']").prop "disabled", yes
+
+    # Enable previous Instrument everywhere
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'] option[value='#{@previous_instrument}']").prop "disabled", no
+
+    # Enable 'None' option as well.
+    $("table.bika-listing-table select.listing_select_entry[field='Instrument'] option[value='']").prop "disabled", no
+
+
+  on_detection_limit_change: (event) =>
+    ###
+     * Eventhandler when the detection limit changed
+    ###
+    console.debug "°°° WorksheetManageResultsView::on_detection_limit_change °°°"
+    $el = $(event.currentTarget)
+
+    defdls = $el.closest("td").find("input[id^='DefaultDLS.']").first().val()
+    resfld = $el.closest("tr").find("input[name^='Result.']")[0]
+    uncfld = $el.closest("tr").find("input[name^='Uncertainty.']")
+
+    defdls = $.parseJSON(defdls)
+    $(resfld).prop "readonly", !defdls.manual
+
+    if $el.val() == "<"
+      $(resfld).val defdls['min']
+      # Inactivate uncertainty?
+      if uncfld.length > 0
+        $(uncfld).val ""
+        $(uncfld).prop "readonly", yes
+        $(uncfld).closest("td").children().hide()
+    else if $el.val() == ">"
+      $(resfld).val defdls["max"]
+      # Inactivate uncertainty?
+      if uncfld.length > 0
+        $(uncfld).val ""
+        $(uncfld).prop "readonly", yes
+        $(uncfld).closest("td").children().hide()
+    else
+      $(resfld).val ""
+      $(resfld).prop "readonly", no
+      # Activate uncertainty?
+      if uncfld.length > 0
+        $(uncfld).val ""
+        $(uncfld).prop "readonly", no
+        $(uncfld).closest("td").children().show()
+
+    # Maybe the result is used in calculations...
+    $(resfld).change()
```

Comment:
Indeed, this js takes into account two aspects about detection limits: the possibility to set a DL manually + relationship between DL and Calculations.

First, Manual entry of Detection Limits, see  https://github.com/bikalims/bika.lims/pull/1496

When "Manual Entry" is set to `True` the user can choose any of the following options from the selection list: `<`, `>` or empty. If the user selects empty, all works as usual: the result entered is the exact result obtained e.g. from the instrument or from the analytical technique used. Indeed, could happen this result is below the detection limit. No problem, `< ldl` will be displayed everywhere (including reports), hiding the submitted-"real" result for that analysis.

On the other hand, if the user selects `<` or `>`, the value he/she introduces is not the exact result obtained, rather he is telling something like "I obtained a result that I don't care about because is below *this* detection limit". Note the difference, in this case, the user does not set a "real" result, rather the detection limit itself. From a technical side, the only difference is that in the latter, the `Result` field from `Analysis`, has a "<" or ">" prepended to the value (not result, rather detection limit) submitted by the user.

Second, Detection Limits + Calculations, see https://github.com/bikalims/bika.lims/pull/1500

.Basically, detection limits can be used in calculations. Consider a formula like:
```python
"Below LDL!" if [AS2.BELOWLDL] else "Not below LDL!"
```
where `AS2` is the keyword of a given Analysis Service.

In summary, if the user changes the value of the DL selection list, we always need to notify that the result has changed, because other Analyses may need to update their result accordingly!.